### PR TITLE
Implementing service catalog management

### PR DESCRIPTION
**Pull request by dperaza4dustbit**
When RegisteredService state change it affects the ServiceCatalog entries. These changes implement that logic.

### CODE REVIEW

File: controllers/registeredservice_controller.go [hunk 1]
```diff
@@ -68,27 +159,11 @@ func (r *RegisteredServiceReconciler) Reconcile(ctx context.Context, req ctrl.Re
...
-		var sc primazaiov1alpha1.ServiceCatalog
-		err = r.Get(ctx, types.NamespacedName{
-			Name:      "primaza-service-catalog",
-			Namespace: req.NamespacedName.Namespace, // "primaza-system",
-		}, &sc)
-		if err != nil {
-			log.Error(err, "Error fetching Service Catalog")
-			return ctrl.Result{}, err
-		}
+		err = r.RemoveServiceFromCatalog(ctx, "primaza-service-catalog", req.NamespacedName.Namespace, req.Name)
```
Comment (filariow) [inline controllers/registeredservice_controller.go:162]: As it is used in multiple places, we may move this to a global unexported constant like 

```
const primazaServiceCatalog = "primaza-service-catalog"
```

File: controllers/registeredservice_controller.go [hunk 2]
```diff
@@ -45,6 +45,97 @@ func ServiceInCatalog(sc primazaiov1alpha1.ServiceCatalog, serviceName string) i
 	return -1
 }
 
+func (r *RegisteredServiceReconciler) RemoveServiceFromCatalog(ctx context.Context, catalogName string, namespace string, serviceName string) error {
```
Comment (filariow) [inline controllers/registeredservice_controller.go:48]: This function can be unexported

```suggestion
func (r *RegisteredServiceReconciler) removeServiceFromCatalog(ctx context.Context, catalogName string, namespace string, serviceName string) error {
```

**Review by filariow — CHANGES_REQUESTED** [review]
In general it LGTM. Just a few nits.

File: controllers/registeredservice_controller.go (hunk 1 above)
Comment (dperaza4dustbit) [reply controllers/registeredservice_controller.go:162]: OK

File: controllers/registeredservice_controller.go [hunk 3]
```diff
@@ -45,6 +45,97 @@ func ServiceInCatalog(sc primazaiov1alpha1.ServiceCatalog, serviceName string) i
...
+	log.Info("Updating Service Catalog")
+	if err := r.Update(ctx, &sc); err != nil {
+		// Service Catalog update failed
+		return err
+	}
+
+	return nil
+}
+
+func (r *RegisteredServiceReconciler) AddServiceToCatalog(ctx context.Context, catalogName string, rs primazaiov1alpha1.RegisteredService) error {
```
[the remark this answers is not in the document]
Comment (dperaza4dustbit) [reply controllers/registeredservice_controller.go]: OK

**Review by filariow — APPROVED** [review]
LGTM

**Pull request merged**